src/me.rs

```diff
@@ -39,6 +38,79 @@ pub fn get_sad(
   sum
 }
 
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[target_feature(enable = "avx2")]
+unsafe fn get_sad_avx2(
+  plane_org: &mut PlaneSlice, plane_ref: &mut PlaneSlice, blk_h: usize,
+  blk_w: usize
+) -> u32 {
+  #[cfg(target_arch = "x86")]
+  use std::arch::x86::*;
+  #[cfg(target_arch = "x86_64")]
+  use std::arch::x86_64::*;
+
+  let mut sum = 0 as u32;
+  let size;
+
+  {
+    let slice_org = plane_org.as_slice_w_width(blk_w);
+    let slice_ref = plane_ref.as_slice_w_width(blk_w);
+    size = ::std::cmp::min(slice_org.len(), slice_ref.len()) as isize;
+
+    let mut ptr_org = slice_org.as_ptr();
+    let mut ptr_ref = slice_ref.as_ptr();
+
+    let mut v32x8sum = _mm256_set1_epi8(0);
+
+    for _r in 0..blk_h {
+      let mut pos = 0;
+      while pos <= size-8 {
```

Comment:
(I think I could replace that by `for pos in (0..=size-8).step_by(8)`, but I'm not sure which is more readable)